pandas/tests/io/excel/test_readers.py

```diff
@@ -432,6 +463,12 @@ def test_read_from_http_url(self, read_ext):
                'pandas/tests/io/data/test1' + read_ext)
         url_table = pd.read_excel(url)
         local_table = pd.read_excel('test1' + read_ext)
+
+        if pd.read_excel.keywords['engine'] == 'openpyxl':
+            # TODO: figure this out. No differences are
```

Comment:
If you can undo this will take a look in CI when it fails

Reply:
Yea this is a very strange error. It gets tripped up on this line of the assertion method:

https://github.com/pandas-dev/pandas/blob/97c85f59aa83564c1e0fb299c9c14d3990d8b189/pandas/util/testing.py#L1257

If you debug there it gets thrown off by the first column ('Unnamed: 0'). All of these fail (essentially triggering the warning):

```python
>>> assert col in right
AssertionError
>>> assert col in right.columns
AssertionError
```

Though these pass:
```python
>>> assert col = right.columns[0]
>>> assert col in right.columns.tolist()
```

I think this might be a deep bug in the hashability of Index objects though not entirely sure.

For this PR maybe just add `by_blocks=True` to these tests with a TODO comment to remove by_blocks

Reply:
Didn't know about the `by_blocks=True`, thanks for pointing that out